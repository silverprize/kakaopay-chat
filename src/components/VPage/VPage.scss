@import '../../scss/variables';
@import '../../scss/mixins';

.page {
  @include flex(column);
  border-radius: $default-border-radius;
  background-color: $content-background-color;
  border: 1px solid $grey-border-color;
  width: 100%;
  height: 100%;
}
