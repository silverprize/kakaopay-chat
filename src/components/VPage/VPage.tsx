import { Component, Vue } from 'vue-property-decorator'
import './VPage.scss'

@Component
export default class VPage extends Vue {
  render() {
    return (
      <div class="page">
        {this.$slots.default}
      </div>
    )
  }
}
