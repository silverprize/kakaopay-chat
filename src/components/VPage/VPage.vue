<template>
  <div class="page">
    <slot />
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import './VPage.scss'

@Component
export default class VPage extends Vue {}
</script>
