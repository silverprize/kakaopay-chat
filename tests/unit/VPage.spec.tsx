import { mount } from '@vue/test-utils'
import VPage from '@/components/VPage/VPage'

describe('VPage', () => {
  it('VPage로 감쌀 콘텐트를 default slot에 넣는다', () => {
    const wrapper = mount(VPage, {
      slots: {
        default: 'foobar',
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
