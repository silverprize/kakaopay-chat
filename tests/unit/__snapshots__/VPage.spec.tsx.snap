// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VPage VPage로 감쌀 콘텐트를 default slot에 넣는다 1`] = `<div class="page">foobar</div>`;
